app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -192,6 +192,14 @@ def find_default_organization_and_or_environment
       end
     end
 
+    def respond_for_show(options = {})
+      if (options.try(:[], :resource).try(:organization_id) && params[:organization_id] &&
```

Comment:
I'd like to suggest a bit of a refactor here, to separate the concerns:

```
    def respond_for_show(options = {})                                                                                                                                                        
      check_resource_organization(options[:resource], params[:organization_id])                                                                                                               
      super                                                                                                                                                                                   
    end                                                                                                                                                                                       
                                                                                                                                                                                              
    def check_resource_organization(resource, organization_id)                                                                                                                                
      return unless resource && resource&.organization_id && organization_id                                                                                                                  
                                                                                                                                                                                              
      if resource.organization_id != organization_id.to_i                                                                                                                                     
        fail HttpErrors::BadRequest, _("The requested resource does not belong to the specified Organization")                                                                                              
      end                                                                                                                                                                                     
    end
```

I think this makes the code more readable by exiting out of `check_resource_organization` if we don't have all of the data to make the comparison. And so the comparison is easier to read! Another distinction: we should raise ::BadRequest instead of ::NotFound because the request is invalid in nature. I also made the message a little 'friendlier'